tests/test_crm.py

```diff
@@ -0,0 +1,538 @@
+
+    # get counters
+    used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_nexthop_group_member_used')
+    avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_nexthop_group_member_available')
+
+    # add route and update available counter
+    ps.set("2.2.2.0/24", fvs)
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_MEMBER_ENTRY', '998')
+
+    time.sleep(60)
+
+    # get counters
+    new_used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_nexthop_group_member_used')
+    new_avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_nexthop_group_member_available')
+
+    assert new_used_counter - used_counter == 2
+    assert avail_counter - new_avail_counter == 2
+
+    # remove route and update available counter
+    ps._del("2.2.2.0/24")
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_NEXT_HOP_GROUP_MEMBER_ENTRY', '1000')
+
+    time.sleep(60)
+
+    # get counters
+    new_used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_nexthop_group_member_used')
+    new_avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_nexthop_group_member_available')
+
+    assert new_used_counter == used_counter
+    assert new_avail_counter == avail_counter
+
+
+def test_CrmFdbEntry(dvs):
+
+    dvs.runcmd("crm config polling interval 1")
+
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_FDB_ENTRY', '1000')
+
+    time.sleep(5*60)
+
+    # get counters
+    used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_used')
+    avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_available')
+
+    app_db = swsscommon.DBConnector(swsscommon.APPL_DB, dvs.redis_sock, 0)
+    cfg_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+
+    # create a FDB entry
+    tbl = swsscommon.ProducerStateTable(app_db, "FDB_TABLE")
+    fvs = swsscommon.FieldValuePairs([("port","Ethernet0"),("type","dynamic")])
+    tbl.set("Vlan2:52-54-00-25-06-E9", fvs)
+
+    # create vlan
+    tbl = swsscommon.Table(cfg_db, "VLAN", "|")
+    fvs = swsscommon.FieldValuePairs([("vlanid", "2")])
+    tbl.set("Vlan2", fvs)
+
+    # create vlan member
+    tbl = swsscommon.Table(cfg_db, "VLAN_MEMBER", "|")
+    fvs = swsscommon.FieldValuePairs([("tagging_mode", "untagged")])
+    tbl.set("Vlan2|Ethernet0", fvs)
+
+    # update available counter
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_FDB_ENTRY', '999')
+
+    time.sleep(60)
+
+    # get counters
+    new_used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_used')
+    new_avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_available')
+
+    assert new_used_counter - used_counter == 1
+    assert avail_counter - new_avail_counter == 1
+
+    # update available counter
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_FDB_ENTRY', '1000')
```

Comment:
Are we not deleting the FDB entry?

Reply:
No, we are not. It is due to problem mentioned in my previous comment for FDB test case:
>volodymyrsamotiy 9 days ago
>Updated the test, for now it does not check updating of "used" counter after removing FDB entry because "FDB remove" operation is not currently working on virtual switch.